platform/string_storage_base.cpp

```diff
@@ -97,6 +97,14 @@ void StringStorageBase::SetValue(string const & key, string && value)
   Save();
 }
 
+void StringStorageBase::Update(std::map<std::string, std::string> const & values)
+{
+  lock_guard<mutex> guard(m_mutex);
+
+  m_values.insert(values.begin(), values.end());
```

Comment:
insert won't overwrite existing elements.

Reply:
Sorry, std:: is always confusing. Updated.